cpp/include/Ice/Properties.h

```diff
@@ -192,6 +192,50 @@ namespace Ice
         std::vector<std::string> _optInPrefixes;
         mutable std::mutex _mutex;
     };
+
+    /// Creates an empty property set.
+    /// @return A new empty property set.
+    ICE_API PropertiesPtr createProperties();
+
+    /// Creates a property set initialized from command-line arguments and a default property set.
+    /// @param seq The command-line arguments. This function parses arguments starting with `--` and one of the
+    /// reserved prefixes (Ice, IceSSL, etc.) as properties and removes these elements from the list. If there is an
+    /// argument starting with `--Ice.Config`, this function loads the specified configuration file. When the same
+    /// property is set in a configuration file and through a command-line argument, the command-line setting takes
+    /// precedence.
+    /// @param defaults Default values for the property set. Settings in configuration files and the arguments override
+    /// these defaults.
+    /// @return A new property set initialized with the properties that were removed from the argument vector.
+    ICE_API PropertiesPtr createProperties(StringSeq& seq, const PropertiesPtr& defaults = nullptr);
```

Comment:
I think this should be deprecated, and document that it just calls the matching properties constructor.

Reply:
We've deprecated them in other languages but not in C++. IIRC it has something to do with the properties constructors.